src/org/traccar/protocol/Arnavi4ProtocolDecoder.java

```
@@ -0,0 +1,170 @@
+                    response = ChannelBuffers.dynamicBuffer(ByteOrder.LITTLE_ENDIAN, 9);
+                    response.writeBytes(new byte[]{0x7B, 0x04, 0x00});
+                    byte[] timeBytes = ByteBuffer.allocate(4).putInt((int) (System.currentTimeMillis() / 1000)).array();
+                    response.writeByte(Checksum.modulo256(timeBytes));
+                    response.writeBytes(timeBytes);
+                    response.writeByte(0x7D);
+
+                } else {
+                    throw new IllegalArgumentException("unsupported header version");
+                }
+
+                channel.write(response);
+            }
+
+            return null;
+        }
+
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress);
+        if (deviceSession == null) {
+            return null;
+        }
+
+        if (startSign == PACKAGE_START_SIGN) {
+
+            List<Position> positions = new LinkedList<>();
+
+            int parcelNumber = buf.readUnsignedByte();
+
+            byte recordStartSign = buf.readByte();
+            while (recordStartSign != PACKAGE_END_SIGN) {
+                switch (recordStartSign) {
+                    case RECORD_PING:
+                    case RECORD_DATA:
+                    case RECORD_TEXT:
+                    case RECORD_FILE:
+                    case RECORD_BINARY:
+                        int length = buf.readUnsignedShort();
+                        long timestamp = buf.readUnsignedInt() * 1000;
+                        ChannelBuffer recordBuf = buf.readBytes(length);
+
+                        if (recordStartSign == RECORD_DATA) {
+                            positions.add(decodePosition(deviceSession, recordBuf, timestamp));
+                        }
+
+                        buf.readUnsignedByte(); // crc
+                        break;
+
+                    default:
+                        throw new IllegalArgumentException("unsupported record type");
```

Should we throw here? Usually we ignore messages if type is unknown. Is it possible?